### PR TITLE
Move baseplate.py sampling and configuration to be more in line with …

### PR DESCRIPTION
**Pull request by trevorriles**
Bring inline with baseplate.go configuration.

- Defaults to sampling at 10 rps.
- Endpoint and service_name are configured via env variables set by infrared.

Closes # 

## 📜 Details
[Design Doc]()

[Jira]()

## ✅ Checks

- [ ] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: baseplate/server/__init__.py [hunk 1]
```diff
@@ -235,66 +229,28 @@ def on_start(self, span: Span, parent_context: Optional[Context] = None) -> None
...
-                )
-            )
-            trace.set_tracer_provider(provider)
-            logger.info("Tracing Configured")
-        else:
-            logger.warning(
-                "Tracing disabled: service_name and endpoint are required fields in the tracing config."
-            )
+    try:
+        sample_rps = int(os.getenv("TRACE_SAMPLER_RPS", 10))
```
Comment (redloaf) [inline baseplate/server/__init__.py]: For things like this, we usually prefix the baseplate env vars with `BASEPLATE_` and use helpers such as `config.DefaultFromEnv`.

**Review by redloaf — CHANGES_REQUESTED** [review]

File: baseplate/server/__init__.py (hunk 1 above)
Comment (trevorriles) [reply baseplate/server/__init__.py]: Awesome, I'll look into get it updated!

**Review by redloaf — APPROVED** [review]

File: baseplate/lib/tracing.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def should_sample(
+        self,
+        parent_context: Optional["Context"],
+        trace_id: int,
+        name: str,
+        kind: Optional[SpanKind] = None,
+        attributes: Attributes = None,
+        links: Optional[Sequence["Link"]] = None,
+        trace_state: Optional["TraceState"] = None,
+    ) -> "SamplingResult":
```
Comment (chriskuehl) [inline baseplate/lib/tracing.py]: The quoting is odd on these, normally it's only needed for self-references (before typing.Self) or when there is weird import-type type construction or stuff like that. If you remove the quoting (and add imports as needed) does it work, or were you getting some error?

```suggestion
        parent_context: Optional[Context],
        trace_id: int,
        name: str,
        kind: Optional[SpanKind] = None,
        attributes: Attributes = None,
        links: Optional[Sequence[Link]] = None,
        trace_state: Optional[TraceState] = None,
    ) -> "SamplingResult":
```

**Review by chriskuehl — APPROVED** [review]

File: baseplate/lib/tracing.py (hunk 2 above)
Comment (trevorriles) [reply baseplate/lib/tracing.py]: I'll update this, PyCharm auto generated with quotes 👍

**Pull request merged**